### PR TITLE
Add an interface defining all the API methods

### PR DESCRIPTION
**Pull request by vdemeester**
Backported from docker/docker project. Makes it easy and readable to know what methods to implement to be considered as a complete docker client 🐰.

/cc @calavera @MHBauer 

🐸

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: client/interface.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	ContainerRename(containerID, newContainerName string) error
+	ContainerResize(options types.ResizeOptions) error
+	ContainerRestart(containerID string, timeout int) error
+	ContainerStatPath(containerID, path string) (types.ContainerPathStat, error)
+	ContainerStats(containerID string, stream bool) (io.ReadCloser, error)
+	ContainerStart(containerID string) error
+	ContainerStop(containerID string, timeout int) error
+	ContainerTop(containerID string, arguments []string) (types.ContainerProcessList, error)
+	ContainerUnpause(containerID string) error
+	ContainerUpdate(containerID string, hostConfig container.HostConfig) error
```
Comment (calavera) [inline client/interface.go]: I just merged #11 and the client won't satisfy this interface anymore :stuck_out_tongue:

Comment (vdemeester) [inline client/interface.go]: Right :smiley_cat:, updating :P

Comment (vdemeester) [inline client/interface.go]: In the near future, it would be cool to somehow check it is :stuck_out_tongue_closed_eyes:

**Comment by calavera** [comment]
@vdemeester tests are welcome :bee: 

LGTM

**Comment by vdemeester** [comment]
@calavera I know :smile: It's on my TODO list as you probably guessed 🐸

**Comment by tiborvass** [comment]
LGTM

**Pull request merged**